### DISCUSSION TITLE
wagmi-cli react plugin `getHookNameError` when two different functions have same name but different casing

### DISCUSSION THREAD

**Discussion by b-pmcg** [wevm/wagmi#2107] [Q&A]
For example the MCD_POT contract (`0x197E90f9FAD81970bA7976f33CbD77088E5D7cf7`) has two different functions, uppercase `Pie` and lowercase `pie`. When trying to generate hooks using the react plugin I'm getting the error `Hook name "useMcdPotPie" must be unique for contract "mcdPot"`

**Comment by tmm** [comment]
Hook names are camelized right now by default. For example, for contracts `FooBar` and `Foo`: If `Foo` has a view function `bar`, then `useFooBar` (`useContract` wrapper) and `useFooBar` (`useContractRead` wrapper, `useContractFunctionRead: true`) will conflict. More info here: https://github.com/wagmi-dev/wagmi/issues/1999#issuecomment-1468531830
